### ISSUE TITLE
Choose file:sendfile/5 by default on Solaris-based systems

### ISSUE THREAD

**Issue by timclassic** [klacke/yaws#224]
`yaws_sendfile_drv.c` has no support for Solaris:
https://github.com/klacke/yaws/blob/master/c_src/yaws_sendfile_drv.c#L159

However, the Yaws implementation of sendfile is still used by default on Solaris.  On my system, the Size variable in the following code becomes `{error, enosys}`:
https://github.com/klacke/yaws/blob/master/src/yaws_server.erl#L3939

I forced Yaws to use file:sendfile/5 during configuration and everything works fine:
```
    %% Use file:sendfile/5 in preference to Yaws sendfile (needed for
    %% Solaris)
    GC2 = ?gc_set_use_erlang_sendfile(GC, true),
    GC3 = ?gc_set_use_yaws_sendfile(GC2, false),
```

Perhaps this should be the default on Solaris?

**Comment by vinoski** [comment]
I may be mistaken but I seem to recall that way back when I originally wrote Yaws sendfile, it did work on Solaris. Getting it to work there should be trivial, but I don't have access to Solaris. I seem to recall that the arguments to the `sendfile` call were different on Solaris than on other platforms, but other than that it worked just fine.

But don't forget: @capflam found a [serious DoS bug in `file:sendfile/5`](http://erlang.org/pipermail/erlang-bugs/2013-October/003818.html), which was fixed in Erlang 17.0. This means that simply setting `file:sendfile/5` as the default for Solaris is a non-starter. Instead, the full fix is to select `file:sendfile/5` when using Erlang versions that don't suffer from that DoS bug, and to also augment Yaws sendfile with Solaris support for use with Erlang versions that have the DoS bug.

**Comment by timclassic** [comment]
@vinoski I will look through the history and see if I can find why Solaris support was dropped, and will also attempt to add it back.

How should the default selection be handled, in general?  Should file:sendfile/5 be preferred on all 17.0 and newer installs, with a fallback to Yaws sendfile for earlier releases?  Is Yaws sendfile always chosen by default currently?

I will read the source to find the answer to that last question, but perhaps you can bootstrap my search.

**Comment by timclassic** [comment]
I have just answered some of my own questions by reading the Yaws release notes.  It appears that file:sendfile/5 was preferred if available until this DoS was found.  I will see if I can simply add a version check to the old mechanism.

**Comment by vinoski** [comment]
Yes, you're on the right track. I believe there's already a version check to see if `file:sendfile/5` is available, so we should be changing that to look for Erlang 17.0 or newer to use `file:sendfile/5`, or fallback to Yaws sendfile otherwise.
